python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -126,6 +126,23 @@ def condition():
     wait_for_condition(condition, timeout=10)
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular", [
+        generate_system_config_map(
+            num_heartbeats_timeout=20, ping_gcs_rpc_server_max_retries=60)
+    ],
+    indirect=True)
+def test_kill_actor_after_gcs_server_restart(ray_start_regular):
+    actor = Increase.remote()
+    result = ray.get(actor.method.remote(1))
+    assert result == 3
+
+    ray.worker._global_node.kill_gcs_server()
+    ray.worker._global_node.start_gcs_server()
+
+    ray.kill(actor)
```

Comment:
This doesn't test anything? We can probably call ray.actors() to see if the states are updated properly?

Reply:
We can not call ray.actors() to see if the states are updated properly.  This bug will cause that `DestroyActor` will not be called, and the status of the actor will still be marked as dead.

Reply:
How is the state of actor still be dead when `DestroyActor` is not called? I think my confusion is from there.

Reply:
The worker is dead, and GCS will calls `ReconstructActor`, which marks the actor state as dead.

Reply:
Gotcha. So, let me clarify if I understood the bug. So the problem is 
- GCS server restarts
- `PollOwnerForActorOutOfScope ` wasn't called except when the state is `DEPENDENCIES_UNREADY `.
- As a result, ray.kill only invoked `ReconstructActor` (but `DestroyActor` wasn't called -> this is a bug).

And your change will make sure `DestroyActor` is called right?

Reply:
I think in this case, we should just remove the test.

Reply:
Oh, actually, I have one idea for testing (it might be a bad test since we should rely on implementation details).

The idea is basically make the actor named actor. And do the following, and make sure name is deleted by calling `ray.get_actor()`. Name is deleted only be `DestroyActor`, so we can verify this works properly.

Reply:
> Gotcha. So, let me clarify if I understood the bug. So the problem is
> 
> * GCS server restarts
> * `PollOwnerForActorOutOfScope ` wasn't called except when the state is `DEPENDENCIES_UNREADY `.
> * As a result, ray.kill only invoked `ReconstructActor` (but `DestroyActor` wasn't called -> this is a bug).
> 
> And your change will make sure `DestroyActor` is called right?

Yeah. you are right. I will remove the test.

Reply:
```
    actor = Increase.options(name="abc").remote()
    result = ray.get(actor.method.remote(1))
    assert result == 3

    ray.worker._global_node.kill_gcs_server()
    ray.worker._global_node.start_gcs_server()

    ray.kill(actor)
    with pytest.raises(ray.exceptions.RayActorError):
        # wait until killed
        actor.method.remote()

    with pytest.raises(ValueError):
        ray.get_actor("abc")

```

Reply:
I think if we change the test name to "test_gcs_restart_named_actor_name_deletion" or something, it makes sense.

Reply:
But named_actors_ is erase in `ReconstructActor`.

Reply:
Actually, since ReconstructActor already remove name entry, we should verify this by del actor instead of ray.kill(actor)

Reply:
yeah I realized that at the same time haha. What about we call `del actor`? This will incur `PollOnwer`..., which uses `DetstroyActor` right?

Reply:
I will try it, thanks.